frontend/deploy.sh

```diff
@@ -0,0 +1,250 @@
+#!/bin/bash
+
+set -eu # add -x to debug
+
+function load_env_variables() {
+    set -a
+    source .env
+    set +a
+}
+
+function checkRequiredParams () {
+    requiredParams=(
+    LOCATION
+    RESOURCE_GROUP
+    SUBSCRIPTION_ID
+    AAD_CLIENT_ID
+    AAD_OBJECT_ID
+    AAD_TENANT_ID
+    )
+    local paramsFile=$1
+    for param in "${requiredParams[@]}"; do
+        local paramValue=$(jq -r .$param < $paramsFile)
+        if [ "null" == "$paramValue" ] || [ -z "$paramValue" ]; then
+            echo "Parameter $param is required, exiting..."
+            exit 1
+        fi
+    done
+}
+
+function populateRequiredParams () {
+    local paramsFile=$1
+    printf "Checking required parameters... "
+    checkRequiredParams $paramsFile
+    # The jq command below sets environment variables based on the key-value pairs in a JSON-formatted file
+    eval $(jq -r 'to_entries | .[] | "export \(.key)=\(.value)"' $paramsFile)
+    printf "Done.\n"
+}
+
+function set_variables() {
+    SUBSCRIPTION_ID=${SUBSCRIPTION_ID:-""}
+    RESOURCE_GROUP=${RESOURCE_GROUP:-""}
+    LOCATION=${LOCATION:-""}
+    AAD_CLIENT_ID=${AAD_CLIENT_ID:-""}
+    AAD_OBJECT_ID=${AAD_OBJECT_ID:-""}
+    AAD_TENANT_ID=${AAD_TENANT_ID:-""}
+    AAD_TOKEN_ISSUER_URL=${AAD_TOKEN_ISSUER_URL:-"https://login.microsoftonline.com/$AAD_TENANT_ID/v2.0"}
+    IMAGE_NAME=${IMAGE_NAME:-"graphrag:frontend"}
+    REGISTRY_NAME=${REGISTRY_NAME:-"${RESOURCE_GROUP}reg"}
+    APP_SERVICE_PLAN=${APP_SERVICE_PLAN:-"${RESOURCE_GROUP}-asp"}
+    WEB_APP=${WEB_APP:-"${RESOURCE_GROUP}-playground"}
+    WEB_APP_IDENTITY=${WEB_APP_IDENTITY:-"${WEB_APP}-identity"}
+    #BACKEND_RESOURCE_GROUP=${BACKEND_RESOURCE_GROUP:-""} # needed for backend outbound vnet integration
+}
+function create_resource_group {
+    az account set --subscription $SUBSCRIPTION_ID
+    az group create --name $RESOURCE_GROUP --location $LOCATION
+}
+
+function create_acr() {
+    az acr create --resource-group $RESOURCE_GROUP \
+    --name $REGISTRY_NAME \
+    --sku Basic \
+    --admin-enabled true
+}
+
+function login_to_acr() {
```

Comment:
I don't believe this step is needed as I was able to comment out the function and script still built and pushed the docker image into the container registry. This step also seemed to be the only place in the script that required docker to be installed.

Reply:
We can remove the `login_to_acr` function if it's not needed.